RoboJackets/robocup-software
# RobotFactoryPosition Layer

## Description
This PR is a refactor of the strategy flow to remove the Coach Node and replace it with the RobotFactoryPosition. This refactor's motivations are explained in the this [design document](https://docs.google.com/document/d/1eVusKIoqb7wYXdVYjSSHk3PGomOQd3RMymBeRynhVuU/edit?usp=sharing).  

Some other notes (Sid):
- this PR deletes the long broken Position control UI. will think about a replacement, but it is noticeably harder in the new paradigm. 

## Associated / Resolved Issue
[ClickUp card 1](https://app.clickup.com/t/863h9ct6x)
[ClickUp card 2](https://app.clickup.com/t/863h9ctf1)

## Design Documents
[Link](https://docs.google.com/document/d/1eVusKIoqb7wYXdVYjSSHk3PGomOQd3RMymBeRynhVuU/edit?usp=sharing)

## Steps to Test
### Test Case 1
1. Ensure that the coach node is not being spun
2. make run-sim

**Expected result:** 4 Defense, 1 Offense, 1 Goalie play shoot only. 

## Key Files to Review
Group 1
 * strategy/agent/position/robot_factory_position.cpp
 * strategy/agent/position/position.hpp

Group 2
 * planning/planner_node.cpp

## Review Checklist
